### PR TITLE
Hughsie/pefile trailing nul

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-bytes.c [hunk 1]
```diff
@@ -181,6 +181,34 @@ fu_bytes_compare(GBytes *bytes1, GBytes *bytes2, GError **error)
...
+/**
+ * fu_bytes_unpad:
+ * @bytes: data blob
+ * @padval: the padding byte to remove, typically 0x00 or 0xFF
+ *
+ * Unpads a GBytes of trailing data.
+ *
+ * Returns: (transfer full): a data blob
+ *
+ * Since: 1.9.21
```
Comment (superm1) [inline libfwupdplugin/fu-bytes.c:193]: we don't have the stream stuff in .21 I thought

Comment (hughsie) [reply libfwupdplugin/fu-bytes.c:193]: Ohh indeed. I was going to backport the symbol, but maybe it doesn't make sense. I'd be fine for 2.0.0

Comment (hughsie) [reply libfwupdplugin/fu-bytes.c:193]: Maybe I'll rework it to do the right [streaming] thing. Otherwise it's a hack on a hack.

Comment (hughsie) [reply libfwupdplugin/fu-bytes.c]: I've reworked it and fixed a few bugs on the journey too.

**Comment by hughsie** [comment]
Interestingly, parsing a big Intel HEX file is better, but not by the masses I thought there would be -- we're still dwarfed but the byte data itself not the streaming overhead.

Screenshot from 2024-06-07 14-07-41

Screenshot from 2024-06-07 14-07-36

**Review by superm1 — APPROVED** [review]

**Pull request merged**